crates/matrix-sdk-crypto/src/identities/user.rs

```diff
@@ -471,7 +612,18 @@ impl OtherUserIdentityData {
     ) -> Result<bool, SignatureError> {
         master_key.verify_subkey(&self_signing_key)?;
 
-        let new = Self::new(master_key, self_signing_key)?;
+        // We update the identity with the new master and self signing key, but we keep
+        // the previous pinned master key.
+        // This identity will have a pin violation until the new master key is pinned
+        // (see [`has_pin_violation`]).
```

Comment:
This won't produce a link since it isn't a doc comment, at least it doesn't for me, for you it does?
```suggestion
        // (see `has_pin_violation()`).
```